samples/led-matrix-weather/Program.cs

```diff
@@ -0,0 +1,274 @@
+using Iot.Device.Graphics;
+using System.Drawing;
+using System.Numerics;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Sockets;
+using System.Net.NetworkInformation;
+
+namespace LedMatrixWeather
+{
+    partial class Program
+    {
+        static Action<RGBLedMatrix> s_scenario = WeatherDemo;
+        static Stopwatch s_showLocalIp = null;
+        static string[] s_ips;
+        static bool s_networkAvailable = false;
+        static Weather s_client;
+        static OpenWeatherResponse s_weatherResponse;
+        static TimeZoneInfo s_timeZonePst = TimeZoneInfo.FindSystemTimeZoneById("America/Los_Angeles");
+
+        static void Main(string[] args)
+        {
+            if (args.Length != 1)
+            {
+                Console.WriteLine("Usage: led-matrix-weather <openweathermap.org API key>");
+                Console.WriteLine();
+                Console.WriteLine("If wrong key is used or there is no network connection");
+                Console.WriteLine("example data will be displayed.");
+                Console.WriteLine();
+                Console.WriteLine("For the first 30 seconds of execution");
+                Console.WriteLine("IP addresses will be displayed instead of data.");
+                return;
+            }
+
+            s_client = new Weather(args[0]);
+
+            UpdateWeather();
+            Task.Run(WeatherUpdater);
+            Task.Run(IpsGetter);
+
+            PinMapping mapping = PinMapping.MatrixBonnetMapping32;
+            RGBLedMatrix matrix = new RGBLedMatrix(mapping, 64, 64, 2, 2);
+
+            Task drawing = Task.Run(() =>
+            {
+                matrix.StartRendering();
+
+                while (true)
+                {
+                    Action<RGBLedMatrix> scenario = s_scenario;
+
+                    if (scenario == null)
+                        break;
+
+                    Stopwatch sw = Stopwatch.StartNew();
+                    scenario(matrix);
+
+                    if (s_scenario != null && sw.ElapsedMilliseconds < 100)
+                    {
+                        Debug.WriteLine("Scenario execution finished in less than 100ms. This is likely due to bug.");
+                    }
+                }
+            });
+
+            try
+            {
+                if (!Console.IsOutputRedirected)
+                {
+                    while (s_scenario != null)
+                    {
+                        switch (Console.ReadKey(intercept: true).Key)
+                        {
+                            case ConsoleKey.Q:
+                            {
+                                s_scenario = null;
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                drawing.Wait();
+            }
+            finally
+            {
+                matrix.Dispose();
+            }
+        }
+
+        static void WeatherUpdater()

```

Comment:
Should this have a cancellationToken? Similar for the  IpsGetter.

Reply:
Don't care for now, the threads will just die with the process anyway and they don't do anything which needs cleanup